packages/realm/package.json

```diff
@@ -73,72 +73,64 @@
         "build:node"
       ]
     },
-    "build": {
-      "dependencies": [
-        "bundle",
-        "build:node",
-        "build:android",
-        "build:ios"
-      ]
-    },
     "bundle": {
       "command": "rollup --config",
       "dependencies": [
-        "../realm-network-transport:build",
-        "build:bindgen:typescript",
-        "build:bindgen:node-wrapper"
+        "../realm-network-transport:bundle",
+        "bindgen:generate:typescript",
+        "bindgen:generate:wrappers"
       ],
       "files": [
         "rollup.config.mjs",
         "src/**/*.ts",
         "tsconfig.json",
         "src/**/tsconfig.json"
       ],
-      "output": [
-        "dist/bundle.*.js",
-        "dist/bundle.*.mjs"
-      ]
+      "output": ["dist"]
     },
-    "build:bindgen": {
-      "command": "cmake-js build --debug --directory ../bindgen",
+    "bindgen:configure": {
+      "command": "cmake-js configure --debug --directory ../bindgen",
+      "dependencies": ["../..:submodules"]
+    },
+    "bindgen:build:node": {
+      "command": "cmake --build ../bindgen/build --target realm-js-node",
+      "dependencies": ["bindgen:configure"]
+    },
```

Comment:
@RedBeard0531 can I get your 👀 on this?

Reply:
I wonder if there are some files and outputs I can use to prevent the configure task to run 🤔

Reply:
Hmm. One option if we are willing to force `cmake-js` to use the `Ninja` generator is to make the output be the `build.ninja`. Then, rather than using `cmake-js configure` you can use `cmake-js build --target build.ninja`. Assuming that wireit treats tasks with no input files as "always dirty", it will always invoke ninja to check if build.ninja needs to be updated or not, and  in the common case, it will very quickly exit with nothing to do.

Integrating multiple build systems is always tricky. Especially if you care about both correctness and incremental builds (which I think we do). The simplest thing will always be to have the "outer" build system unconditionally invoke the "inner" build system and let it handle deciding if it needs to do anything or not. Ideally the outer build system has a way to detect if the output files have changed *after* running the inner build system, so that it can only rerun downstream tasks when needed. We should only deviate from the "simple" solution if it brings enough benefit to outweigh the complexity and risks.

Reply:
> always be to have the "outer" build system unconditionally invoke the "inner" build system

As I don't declare files and outputs, this is the current (default) behaviour.